### PR TITLE
[10.0][IMP] medical: Add abstract entity model + views

### PR DESCRIPTION
**Pull request by lasley**
This is contained in OCA#119, and is the only difference between it and our current release/10.0

### CODE REVIEW

**Comment by lasley** [comment]
This can be reviewed independently from the OCA PR. I'm about to start making the downstream changes

File: medical/views/medical_patient.xml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                           domain="[('customer','=',True)]"
+                           />
+                </group>
+                <group>
+                    <field name="birthdate_date" />
+                    <field name="date_death" />
+                    <field name="age" />
+                </group>
+            </xpath>
+            <xpath expr="//img[@name='kanban_image_default_person']" position="replace">
```
Comment (tedsalmon) [inline medical/views/medical_patient.xml]: Lint is complaining about this replace without a priority.
```
Dangerous use of "replace" from view with priority 0 < 99
```

File: medical/views/medical_patient.xml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                    <strong>DOB:</strong> <field name="birthdate_date" />
+                </li>
+                <li name="field_age_gender" t-if="record.birthdate_date.raw_value &amp;&amp; record.gender.raw_value">
+                    <field name="age" /> - <field name="gender" />
+                </li>
+                <li name="field_gender" t-if="! record.birthdate_date.raw_value &amp;&amp; record.gender.raw_value">
+                    <field name="gender" />
+                </li>
+            </xpath>
+            <xpath expr="//img[@name='kanban_image_default_person']" position="replace">
```
Comment (tedsalmon) [inline medical/views/medical_patient.xml]: Lint is complaining about this replace without a priority.
```
Dangerous use of "replace" from view with priority 0 < 99
```

File: medical/security/ir.model.access.csv [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
+doctor_medical_patient,doctor_medical_patient,model_medical_patient,group_medical_doctor,1,1,1,0
+user_medical_patient,user_medical_patient,model_medical_patient,group_medical_user,1,1,1,0
+assistant_res_partner,assistant_res_partner,model_res_partner,group_medical_assistant,1,1,1,0
+center_manager_res_partner,center_manager_res_partner,model_res_partner,group_medical_center_manager,1,1,1,0
+doctor_res_partner,doctor_res_partner,model_res_partner,group_medical_doctor,1,1,1,0
+user_res_partner,user_res_partner,model_res_partner,group_medical_user,1,1,0,0
+assistant_medical_abstract_entity,assistant_medical_abstract_entity,model_medical_abstract_entity,group_medical_assistant,1,1,1,0
+center_manager_medical_abstract_entity,center_manager_medical_abstract_entity,model_medical_abstract_entity,group_medical_center_manager,1,1,1,0
+doctor_medical_abstract_entity,doctor_medical_abstract_entity,model_medical_abstract_entity,group_medical_doctor,1,1,1,0
+user_medical_abstract_entity,user_medical_abstract_entity,model_medical_abstract_entity,group_medical_user,1,1,1,0
```
Comment (tedsalmon) [inline medical/security/ir.model.access.csv:17]: No newline at EOF

**Review by tedsalmon — APPROVED** [review]
Minor details commented on. Otherwise, LGTM! 👍

**Comment by lasley** [comment]
Note that this is technically broken on Runbot, but fixed by uninstalling website. We have LABS-373 for that so we can push forward here.

**Pull request merged**